app/routers/quizzes.py

```diff
@@ -92,6 +93,35 @@ async def get_quiz(quiz_id: str):
     quiz_collection = client.quiz.quizzes
     if (quiz := quiz_collection.find_one({"_id": quiz_id})) is not None:
         update_quiz_for_backwards_compatibility(quiz_collection, quiz_id, quiz)
+
+        if quiz["metadata"]["quiz_type"] == QuizType.omr.value:
```

Comment:
also put null checks for these keys no?
Like if `metadata` is not in quiz or `quiz_type` is not in `quiz["metadata"]`? Just to be sure?